holoviews/plotting/bokeh/element.py

```diff
@@ -1919,6 +1928,30 @@ def initialize_plot(self, ranges=None, plot=None, plots=None, source=None):
 
         return plot
 
+    def _apply_hard_bounds(self, element, ranges):
+        """
+        Apply hard bounds to the x and y ranges of the plot. If xlim/ylim is set, limit the
+        initial viewable range to xlim/ylim, but allow navigation up to the abs max between
+        the data range and xlim/ylim. If dim range is set (e.g. via redim.range), enforce
+        as hard bounds.
+
+        """
+
+        def validate_bound(bound):
+            return bound if util.isfinite(bound) else None
+
+        min_extent_x, min_extent_y, max_extent_x, max_extent_y = map(validate_bound, self.get_extents(element, ranges, range_type='combined', lims_as_soft_ranges=True))
+
+        def set_bounds(axis, min_extent, max_extent):
+            """Set the bounds for a given axis, using None if both extents are None or identical"""
+            if min_extent == max_extent:
+                self.handles[axis].bounds = None
+            else:
+                self.handles[axis].bounds = (min_extent, max_extent) if min_extent is not None or max_extent is not None else None
```

Comment:
```suggestion
                self.handles[axis].bounds = (min_extent, max_extent) if min_extent is not None else None
```

If `min_extent` is not None here `max_extent` isn't either because of `min_extent == max_extent`.

Reply:
Actually, I think we can take this a step further.. if at least one of min_extent, max_extent is not None then we want to return the tuple.. so I think it can just be:

```python
if min_extent == max_extent:
    self.handles[axis].bounds = None
else:
    self.handles[axis].bounds = (min_extent, max_extent)
```

Reply:
If that is the case, you can make it into a one-liner

Reply:
pushed this change, let me know if you disagree